documentation/Glossary.md

```diff
@@ -0,0 +1,149 @@
+
+See also: [Projection](#projection).
+
+## ColumnRole
+
+Meta-information about a [Column](#column) that defines the Column's role in data presentation. For example, a Column that contains a `Timestamp` whose value indicates when an event occured relative to the start of a [DataSource](#datasource) may be marked as a "start time" Column.
+
+## CompositeDataCooker
+
+A [DataCooker](#datacooker) that receives input solely from other DataCookers.
+
+## CustomDataProcessor
+
+Component responsible for processing a [DataSource](#datasource). For example, a CustomDataProcessor may be responsible for parsing lines in a `.txt` file into `LineItem` objects that later get consumed by a [DataCooker](#datacooker) or [Table](#table).
+
+In most situations, a CustomDataProcessor will further delegate the responsibility of processing a DataSource to a [SourceParser](#sourceparser).
+
+A CustomDataProcessor is responsible for building any [Simple Tables](#simple-table) discovered by the [ProcessingSource](#processingsource) that created it.
+
+## DataCooker
+
+Component of a [Processing Pipeline](#processing-pipeline) that consumes data from one or more sources and optionally outputs new data for other components to consume. Conceptually, a DataCooker "cooks" data from one type to another.
+
+See also: [SourceDataCooker](#sourcedatacooker) and [CompositeDataCooker](#compositedatacooker).
+
+## DataSource
+
+An input to be processed by a [CustomDataProcessor](#customdataprocessor). In most situations, this is a `FileDataSource` that contains the full path to a file specified by a user. 
+
+## DynamicTable
+
+A [Table](#table) that is built outside of the standard [table building cycle](#table-building-cycle). DynamicTables are commonly built in response to a [TableCommand](#tablecommand) being invoked.
+
+## Plugin
+
+A collection of one-or-more [ProcessingSources](#processingsource) that collectively create zero-or-more [Tables](#table). Plugins also consist of zero-or-more [DataCookers](#datacooker).
+
+## Processing Pipeline
+
+A collection of [DataCookers](#datacooker) and [SourceParsers](#sourceparser) that define a structured flow of data.
+
+## ProcessingSource
+
+An entrypoint for a [Plugin](#plugin). A ProcessingSource
+1. Declares a human-readable name and description
+2. Defines the [DataSource(s)](#datasource) it supports
+3. Creates the [CustomDataProcessor](#customdataprocessor) that will process instances of supported DataSources
+4. Advertises the [Table(s)](#table) that may be built in response to DataSources being processed
+
+## Projection
+
+A function that maps a row index for a [Table](#table) to a piece of data. Conceptually, a Projection is combined with a [ColumnConfiguration](#columnconfiguration) to complete define a Table's column: the ColumnConfiguration defines the name and metadata about a column, and its associated Projection defines the column's data.
```

Comment:
>[Table](#table) [](http://example.com/codeflow?start=39&length=15)

do we need to mention columns here?

Reply:
yep - I wrote this one before Column and forgot to update. Thanks!